dev/archery/archery/linking.py

```diff
@@ -61,9 +63,83 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _remove_weak_symbols(self, symbol_info):
+        return [line for line in symbol_info if not line.endswith(" w")]
```

Comment:
This does not cover all kinds of weak symbols, see `nm` manpage for details.

Reply:
Sorry I didn't catch this earlier, are you referring to `w` and `v`? Or is there more to it?

Reply:
There's `v`, `V`, `w` and `W` according to the manpage on my Ubuntu 22.04 computer.

Reply:
You can use `str.endswith` with a tuple of suffixes:
`line.endswith(" v", " V", " w", " W")`

Reply:
I will check this out. Thank you.

Reply:
it should be `line.endswith((" v", " V", " w", " W"))` ?